moto/glue/glue_schema_registry_utils.py

```diff
@@ -0,0 +1,411 @@
+import re
+import json
+
+from .glue_schema_registry_constants import (
+    MAX_REGISTRY_NAME_LENGTH,
+    RESOURCE_NAME_PATTERN,
+    MAX_ARN_LENGTH,
+    ARN_PATTERN,
+    MAX_DESCRIPTION_LENGTH,
+    DESCRIPTION_PATTERN,
+    MAX_SCHEMA_NAME_LENGTH,
+    DEFAULT_REGISTRY_NAME,
+    REGISTRY_NAME,
+    REGISTRY_ARN,
+    SCHEMA_NAME,
+    SCHEMA_ARN,
+    MAX_TAGS_ALLOWED,
+    MAX_SCHEMA_DEFINITION_LENGTH,
+    SCHEMA_DEFINITION,
+    MAX_SCHEMAS_ALLOWED,
+    MAX_SCHEMA_VERSIONS_ALLOWED,
+    MAX_REGISTRIES_ALLOWED,
+    SCHEMA_VERSION_ID_PATTERN,
+    SCHEMA_VERSION_ID,
+    VERSION_NUMBER,
+    LATEST_VERSION,
+    METADATA_VALUE,
+    METADATA_KEY,
+    MAX_SCHEMA_VERSION_METADATA_LENGTH,
+    SCHEMA_VERSION_METADATA_PATTERN,
+    MAX_SCHEMA_VERSION_METADATA_ALLOWED,
+)
+
+from .exceptions import (
+    ResourceNameTooLongException,
+    ParamValueContainsInvalidCharactersException,
+    InvalidSchemaDefinitionException,
+    InvalidRegistryIdBothParamsProvidedException,
+    RegistryNotFoundException,
+    InvalidSchemaIdBothParamsProvidedException,
+    InvalidSchemaIdNotProvidedException,
+    SchemaNotFoundException,
+    InvalidDataFormatException,
+    InvalidCompatibilityException,
+    InvalidNumberOfTagsException,
+    GeneralGSRAlreadyExistsException,
+    GeneralResourceNumberLimitExceededException,
+    DisabledCompatibilityVersioningException,
+    InvalidSchemaVersionNumberBothParamsProvidedException,
+    InvalidSchemaVersionIdProvidedWithOtherParamsException,
+    InvalidSchemaVersionNumberNotProvidedException,
+    SchemaVersionMetadataLimitExceededException,
+)
+
+
+def validate_registry_name_pattern_and_length(param_value):
+    param_name = "registryName"
+    max_name_length = MAX_REGISTRY_NAME_LENGTH
```

Comment:
Why assign these constants to a local variable? Can't they be inlined into the function call that immediately follows?

So in this case it would become this instead:

```py
validate_param_pattern_and_length(
  param_value, 
  param_name = "registryName", 
  max_name_length=MAX_REGISTRY_NAME_LENGTH, 
  pattern=RESOURCE_NAME_PATTERN
)
```

Reply:
Thanks for pointing that out.

Reply:
Doing this would be Redefining name `glue_client` or `client` from outer scope. And this is not advisable.